src/traces/bar/layout_defaults.js

```diff
@@ -46,7 +46,10 @@ module.exports = function(layoutIn, layoutOut, fullData) {
         }
     }
 
-    if(!hasBars) return;
+    if(!hasBars) {
+        delete layoutOut.barmode;
```

Comment:
`++` a jasmine test in `bar_test.js` plz :smirk:

Reply:
a jasmine test that checks `_fullLayout` doesn't have a `barmode`?

Reply:
yeah, that and the converse. Similar to https://github.com/plotly/plotly.js/blob/ef932cfad9eb35fb7986a3740fdb42464d6e8b7f/test/jasmine/tests/bar_test.js#L214-L228

Reply:
Done in bc3890b